WIP: "199 sns logging"

fhilg opened the pull request:
## Type of this PR 

- [ ] Bug fix
- [x] Enhancement

## Description
- SNS log level changed on message reception
- log message on radio configuration added to SNS

## Issue(s) related to this PR

 * internal issue 199
 
 
## Affected parts of the online documentation
documentation not affected

## Definition of Done

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [ ] All commits are signed-off (`-s`) with your mail address of your Eclipse Account.
- [x] `origin/main` has been merged into your Fork.
- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [x] There are no new SpotBugs warnings. 
- [x] Coding guidelines have been observed (see CONTRIBUTING.md).
- [ ] All tests pass.

## Special notes to reviewer

fhilg commented:
## Log Level Change: Message Reception
**example log message:**
```
TRACE Ns3Ambassador - Receive V2XMessage : Id(146) on Node rsu_2 at Time=105048712731
```

**Pro:**
- less output
- output of applications may be redundant (same kind of information on another level may be defined)
- reception may occur as often as vehicle move event, which is already at level TRACE

**Con:**
- DEBUG is already verbose, the TransmissionSimulator should be changed, too, **or** the log level change should be reverted
- If other components are equally verbose at level DEBUG, this change can be considered as consistency break.


## Examples
Reception at level TRACE
### Level INFO
```
2021-08-12 12:10:20,787 INFO  SnsAmbassador - Start Simulation
2021-08-12 12:10:21,165 INFO  SnsAmbassador - Initialized SNS
2021-08-12 12:10:21,189 INFO  SnsAmbassador - Added RSU id=rsu_0 @time=0.000,000,000 s
2021-08-12 12:10:21,189 INFO  SnsAmbassador - Added RSU id=rsu_1 @time=0.000,000,000 s
2021-08-12 12:10:21,189 INFO  SnsAmbassador - Added RSU id=rsu_2 @time=0.000,000,000 s
2021-08-12 12:10:22,384 WARN  SnsAmbassador - Node rsu_0 is not configured with a distance value. Using global singlehop radius from SNS configuration.
2021-08-12 12:10:22,384 INFO  SnsAmbassador - Radio configured in mode SINGLE with communication radius 454.7354659822783 for node id=rsu_0 @time=0.000,000,000 s
2021-08-12 12:10:22,384 WARN  SnsAmbassador - Node rsu_1 is not configured with a distance value. Using global singlehop radius from SNS configuration.
2021-08-12 12:10:22,384 INFO  SnsAmbassador - Radio configured in mode SINGLE with communication radius 454.7354659822783 for node id=rsu_1 @time=0.000,000,000 s
2021-08-12 12:10:22,384 WARN  SnsAmbassador - Node rsu_2 is not configured with a distance value. Using global singlehop radius from SNS configuration.
2021-08-12 12:10:22,384 INFO  SnsAmbassador - Radio configured in mode SINGLE with communication radius 454.7354659822783 for node id=rsu_2 @time=0.000,000,000 s
2021-08-12 12:10:22,405 INFO  SnsAmbassador - Radio configured in mode SINGLE with communication radius 321.54653164209356 for node id=veh_0 @time=2.000,000,000 s
2021-08-12 12:10:22,488 INFO  SnsAmbassador - Radio configured in mode SINGLE with communication radius 454.7354659822783 for node id=veh_1 @time=61.000,000,000 s
2021-08-12 12:10:22,599 INFO  SnsAmbassador - Removed Vehicle id=veh_0 @time=177.000,000,000 s
2021-08-12 12:10:22,636 INFO  SnsAmbassador - Removed Vehicle id=veh_1 @time=236.000,000,000 s
```


### Level TRACE (excerpt)
```
...
2021-08-12 12:14:54,282 DEBUG TransmissionSimulator - Send v2xMessage.id=287 from node=veh_0 as Topocast (singlehop) @time=176.021,000,000 s
2021-08-12 12:14:54,282 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
2021-08-12 12:14:54,282 DEBUG TransmissionSimulator - Send v2xMessage.id=288 from node=veh_1 as Topocast (singlehop) @time=176.035,000,000 s
2021-08-12 12:14:54,282 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
2021-08-12 12:14:54,283 INFO  SnsAmbassador - Removed Vehicle id=veh_0 @time=177.000,000,000 s
2021-08-12 12:14:54,284 DEBUG TransmissionSimulator - Send v2xMessage.id=289 from node=veh_1 as Topocast (singlehop) @time=177.035,000,000 s
2021-08-12 12:14:54,284 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
2021-08-12 12:14:54,286 DEBUG TransmissionSimulator - Send v2xMessage.id=290 from node=veh_1 as Topocast (singlehop) @time=178.035,000,000 s
2021-08-12 12:14:54,286 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
...
```


### Level TRACE (excerpt)
```
...
2021-08-12 12:11:28,148 TRACE SnsAmbassador - Moved Vehicle id=veh_1 to position=GeoPoint{lat=52.513751,lon=13.340482,alt=0.00} @time=176.000,000,000 s
2021-08-12 12:11:28,148 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 176000000000
2021-08-12 12:11:28,149 DEBUG TransmissionSimulator - Send v2xMessage.id=287 from node=veh_0 as Topocast (singlehop) @time=176.021,000,000 s
2021-08-12 12:11:28,149 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
2021-08-12 12:11:28,149 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 176021000000
2021-08-12 12:11:28,149 DEBUG TransmissionSimulator - Send v2xMessage.id=288 from node=veh_1 as Topocast (singlehop) @time=176.035,000,000 s
2021-08-12 12:11:28,150 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
2021-08-12 12:11:28,150 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 176035000000
2021-08-12 12:11:28,151 TRACE SnsAmbassador - Moved Vehicle id=veh_1 to position=GeoPoint{lat=52.513761,lon=13.340628,alt=0.00} @time=177.000,000,000 s
2021-08-12 12:11:28,151 INFO  SnsAmbassador - Removed Vehicle id=veh_0 @time=177.000,000,000 s
2021-08-12 12:11:28,151 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 177000000000
2021-08-12 12:11:28,152 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 177000000000
2021-08-12 12:11:28,152 DEBUG TransmissionSimulator - Send v2xMessage.id=289 from node=veh_1 as Topocast (singlehop) @time=177.035,000,000 s
2021-08-12 12:11:28,152 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
2021-08-12 12:11:28,152 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 177035000000
2021-08-12 12:11:28,153 TRACE SnsAmbassador - Moved Vehicle id=veh_1 to position=GeoPoint{lat=52.513771,lon=13.340775,alt=0.00} @time=178.000,000,000 s
2021-08-12 12:11:28,153 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 178000000000
2021-08-12 12:11:28,154 DEBUG TransmissionSimulator - Send v2xMessage.id=290 from node=veh_1 as Topocast (singlehop) @time=178.035,000,000 s
2021-08-12 12:11:28,154 DEBUG TransmissionSimulator - Addressed nodes in destination area={}
2021-08-12 12:11:28,154 TRACE SnsAmbassador - processTimeAdvanceGrant(time); time: 178035000000
2021-08-12 12:11:28,154 TRACE SnsAmbassador - Moved Vehicle id=veh_1 to position=GeoPoint{lat=52.513781,lon=13.340921,alt=0.00} @time=179.000,000,000 s
...
```

fhilg commented:
Communication logs of scenario `AdhocCommunication` with MOSAIC's debug level set to "debug" or "info" or "trace":
- [199 SNS DEBUG.zip](https://github.com/eclipse/mosaic/files/6976753/199.SNS.DEBUG.zip)
- [199 SNS INFO.zip](https://github.com/eclipse/mosaic/files/6976755/199.SNS.INFO.zip)
- [199 SNS TRACE.zip](https://github.com/eclipse/mosaic/files/6976757/199.SNS.TRACE.zip)

Federate debug levels are not changed (ns-3 and OMNeT++).

kschrab commented:
```
2021-08-12 17:40:13,910 DEBUG TransmissionSimulator - Addressed nodes in destination area={rsu_0=org.eclipse.mosaic.fed.sns.ambassador.SimulationNode@7f8920f8, rsu_1=org.eclipse.mosaic.fed.sns.ambassador.SimulationNode@13df6802}
```

I would propose that you implement `toString` in class `SimulationNode` for better printing here:

```java
public String toString() {
    return "(" + position.x + ", " + position.y + ")";
}

kschrab commented:
Comments to AbstractNetworkAmbassador log:

AbstractNetwork:

Should be TRACE:
```
2021-08-12 17:40:26,016 DEBUG OmnetppAmbassador - ProcessTimeAdvanceGrant at time=0
2021-08-12 17:40:26,122 DEBUG OmnetppAmbassador - End ProcessTimeAdvanceGrant at: 0
```

Not helpful (delete)
```
2021-08-12 17:40:26,138 DEBUG OmnetppAmbassador - Add Vehicles at first movement
```

Not helpful / redundan (delete)
```
2021-08-12 17:40:26,235 DEBUG OmnetppAmbassador - Update vehicle positions.
```

This information is not helpful. It is sufficient to just have a message if the message is NOT applicable (delete)
```
2021-08-12 17:40:26,188 DEBUG OmnetppAmbassador - This V2XMessage is applicable for this network simulator. Send this message. V2XMessage.id=0
```

Merge into one line
```
2021-08-12 17:40:26,218 DEBUG OmnetppAmbassador - UpdateNode : ID[int=veh_0, ext=3]
2021-08-12 17:40:26,218 DEBUG OmnetppAmbassador - Pos: x(491.59299171292923) y(136.7464899286799) Geo: GeoPoint{lat=52.512619,lon=13.323940,alt=0.00}
```

Information about receiving/delivering a message is completely missing, should be one line in DEBUG

fhilg commented:
new logs of SNS (same scenario):
- [INFO Communication.log](https://github.com/eclipse/mosaic/files/7013245/INFO.Communication.log)
- [DEBUG Communication.log](https://github.com/eclipse/mosaic/files/7013243/DEBUG.Communication.log)
- [TRACE Communication.log](https://github.com/eclipse/mosaic/files/7013246/TRACE.Communication.log)

kschrab commented:
> Comments to AbstractNetworkAmbassador log:
> 
> AbstractNetwork:
> 
> * [ ]  Should be TRACE:
>   ```
>   2021-08-12 17:40:26,016 DEBUG OmnetppAmbassador - ProcessTimeAdvanceGrant at time=0
>   2021-08-12 17:40:26,122 DEBUG OmnetppAmbassador - End ProcessTimeAdvanceGrant at: 0
>   ```
> * [ ]  Not helpful (delete)
>   ```
>   2021-08-12 17:40:26,138 DEBUG OmnetppAmbassador - Add Vehicles at first movement
>   ```
> * [ ]  Not helpful / redundan (delete)
>   ```
>   2021-08-12 17:40:26,235 DEBUG OmnetppAmbassador - Update vehicle positions.
>   ```
> * [ ]  This information is not helpful. It is sufficient to just have a message if the message is NOT applicable (delete)
>   ```
>   2021-08-12 17:40:26,188 DEBUG OmnetppAmbassador - This V2XMessage is applicable for this network simulator. Send this message. V2XMessage.id=0
>   ```
> * [ ]  Merge into one line
>   ```
>   2021-08-12 17:40:26,218 DEBUG OmnetppAmbassador - UpdateNode : ID[int=veh_0, ext=3]
>   2021-08-12 17:40:26,218 DEBUG OmnetppAmbassador - Pos: x(491.59299171292923) y(136.7464899286799) Geo: GeoPoint{lat=52.512619,lon=13.323940,alt=0.00}
>   ```
> * [ ]  Should be INFO (similar to SNS)
>   ```
>   2021-08-12 17:40:26,147 DEBUG OmnetppAmbassador - AdHocCommunicationConfiguration: from node ID[int=veh_0, ext=3], at time = 2000000000 channels: [CCH,null|null,null]
>   ```

@fhilg In the changes tab I cannot see that you addressed all of these issues regarding logging of AbstractNetworkAmbassador. What is still open here?

fhilg commented:
> > Comments to AbstractNetworkAmbassador log:
> > AbstractNetwork:
> > 
> > * [ ]   Should be TRACE:
> >   ```
> >   2021-08-12 17:40:26,016 DEBUG OmnetppAmbassador - ProcessTimeAdvanceGrant at time=0
> >   2021-08-12 17:40:26,122 DEBUG OmnetppAmbassador - End ProcessTimeAdvanceGrant at: 0
> >   ```
see [bf90ba3](https://github.com/eclipse/mosaic/pull/138/commits/bf90ba3f9d927b96ee639d827d27021416e1d3c5)
> > * [ ]   Not helpful (delete)
> >   ```
> >   2021-08-12 17:40:26,138 DEBUG OmnetppAmbassador - Add Vehicles at first movement
> >   ```
see [1caa0d3](https://github.com/eclipse/mosaic/pull/138/commits/1caa0d3142ffe0112997ebbe35d154406945ef1a)
> > * [ ]   Not helpful / redundan (delete)
> >   ```
> >   2021-08-12 17:40:26,235 DEBUG OmnetppAmbassador - Update vehicle positions.
> >   ```
see [1caa0d3](https://github.com/eclipse/mosaic/pull/138/commits/1caa0d3142ffe0112997ebbe35d154406945ef1a)
> > * [ ]   This information is not helpful. It is sufficient to just have a message if the message is NOT applicable (delete)
> >   ```
see [1caa0d3](https://github.com/eclipse/mosaic/pull/138/commits/1caa0d3142ffe0112997ebbe35d154406945ef1a)
> >   2021-08-12 17:40:26,188 DEBUG OmnetppAmbassador - This V2XMessage is applicable for this network simulator. Send this message. V2XMessage.id=0
> >   ```
> > * [ ]   Merge into one line
> >   ```
> >   2021-08-12 17:40:26,218 DEBUG OmnetppAmbassador - UpdateNode : ID[int=veh_0, ext=3]
> >   2021-08-12 17:40:26,218 DEBUG OmnetppAmbassador - Pos: x(491.59299171292923) y(136.7464899286799) Geo: GeoPoint{lat=52.512619,lon=13.323940,alt=0.00}
> >   ```
see [831b817](https://github.com/eclipse/mosaic/pull/138/commits/831b817de7333c9665a624c15511051adf415b58)
> > * [ ]   Should be INFO (similar to SNS)
> >   ```
> >   2021-08-12 17:40:26,147 DEBUG OmnetppAmbassador - AdHocCommunicationConfiguration: from node ID[int=veh_0, ext=3], at time = 2000000000 channels: [CCH,null|null,null]
> >   ```
see [867e271](https://github.com/eclipse/mosaic/pull/138/commits/867e2713ccbae15b11bb8c34edd9b88d77c4e72e),
changed to trace, therefore this is still open
> 
> @fhilg In the changes tab I cannot see that you addressed all of these issues regarding logging of AbstractNetworkAmbassador. What is still open here?

kschrab commented on fed/mosaic-sns/src/main/java/org/eclipse/mosaic/fed/sns/ambassador/SnsAmbassador.java:
```diff
@@ -75,12 +75,13 @@
     @Override
     public void initialize(final long startTime, final long endTime) throws InternalFederateException {
         super.initialize(startTime, endTime);
+        this.log.info("Init simulation with startTime={}, stopTime={}", startTime, endTime);
```
better call it `endTime`

fhilg replied:
done
https://github.com/fhilg/mosaic/commit/a6a043557a81ce10b87586eccd5dc9c23c792c3b

kschrab commented on fed/mosaic-sns/src/main/java/org/eclipse/mosaic/fed/sns/ambassador/TransmissionSimulator.java:
```diff
@@ -41,7 +41,7 @@
 public class TransmissionSimulator {
 
-    private final static Logger log = LoggerFactory.getLogger(TransmissionSimulator.class);
+    private final static Logger log = LoggerFactory.getLogger(SnsAmbassador.class);
```
I'd prefer that log messages generated in this class will also have the class name as prefix. Was this proposed by @rprotzmann ?

fhilg replied:
Exactly, @rprotzmann requested this change in our yesterday's meeting.

rprotzmann replied:
Yes, I would prefer this solution - otherwise sendV2XMessage is logged with TransmissionSimulator (which actually calculated the whole transmission from sending to receiving), but recvV2XMessage is logged again by SnsAmbassador (because it is previously splitted into several invidual interactions

kschrab commented on fed/mosaic-sns/src/main/java/org/eclipse/mosaic/fed/sns/ambassador/SnsAmbassador.java:
```diff
@@ -185,13 +186,17 @@ private void process(AdHocCommunicationConfiguration interaction) {
                 } else {
+                    // Vehicle has not been simulated by the SNS, yet. Radio is configured for the first time.
                     registeredVehicles.put(nodeId, communicationRadius);
+                    log.info("Added Vehicle id={} @time={}", nodeId, TIME.format(interaction.getTime()));
```
I'm not sure if this is correct. I would print this message when the vehicle goes "online" and not when it is "pre-registered". So moving this line to 192 would be correct, wouldn't it?

fhilg replied:
I thought this was intended. Maybe we should discuss it with @rprotzmann .

kschrab replied:
Change DEBUG and write "Pre-register radio configuration for vehicle id=..."

kschrab commented on fed/mosaic-sns/src/main/java/org/eclipse/mosaic/fed/sns/ambassador/SnsAmbassador.java:
```diff
@@ -127,30 +128,30 @@ private void process(RsuRegistration interaction) {
         for (VehicleData added : interaction.getAdded()) {
             if (addOrUpdateVehicle(added)) {
-                log.info("Added Vehicle id={} @time={}", added.getName(), TIME.format(interaction.getTime()));
+                log.info("Added Vehicle id={} position={} @time={}", added.getName(), added.getPosition(), TIME.format(interaction.getTime()));
```
You should move this line to `addOrUpdateVehicle` where the method `createOnlineNode` is called, I guess

kschrab replied:
But that's exactly the case we're not interested in. We don't care for vehicles which are never considered by the communication simulation (e.g. background traffic).

fhilg replied:
Oh, the case I mentioned was not applicable, here, sorry.

kschrab replied:
@fhilg Have you worked on this? This line is currently missing in the log files.

The line should be move to `addOrUpdateVehicle` where the method `createOnlineNode` is called

fhilg replied:
No, I haven't moved this line because
- it was a guess, and
- there was no decision made in our last meeting.

kschrab replied:
> there was no decision made in our last meeting.

It actually was decided ;) maybe it was not too clear. Please just do what I've proposed and see if it works

fhilg commented:
See new logs in pull request of [OMNeT++ federate](https://github.com/mosaic-addons/omnetpp-federate/pull/4)

rprotzmann commented on lib/mosaic-network/src/main/java/org/eclipse/mosaic/lib/coupling/AbstractNetworkAmbassador.java:
```diff
@@ -438,7 +445,6 @@ private synchronized void process(VehicleUpdates interaction) throws InternalFed
-                log.debug("Add Vehicles at first movement");
                 List<VehicleData> addedVehicles = interaction.getAdded();
                 Comparator<UnitData> comp = new UnitNameComparator();
                 addedVehicles.sort(comp);
```
Line 456 not needed to be warn, just debug:
`log.warn("Vehicle with ID {} is not in the registered list (no config arrived yet), ignoring entry.", vi.getName());`
better:
`log.debug("Vehicle with ID {} is not in the registered list (no config arrived yet)", vi.getName());`

fhilg replied:
done in https://github.com/eclipse/mosaic/pull/138/commits/13dbff3280132b74ded0334bed0236dae43794ad

fhilg replied:
done in https://github.com/eclipse/mosaic/pull/138/commits/9523213c616dfcfd3fe1b1ae2d70dbe890f20ce8

rprotzmann commented on lib/mosaic-network/src/main/java/org/eclipse/mosaic/lib/coupling/AbstractNetworkAmbassador.java:
```diff
@@ -562,18 +569,17 @@ private synchronized void process(V2xMessageTransmission interaction) throws Int
+                        "sendV2XMessage: id={} from node ID[int={} , ext={}] channel:{} type:{} time={}",
                         interaction.getMessageId(),
-                        sac.getSourceName(), sourceId, dac.getAdhocChannelId(), interaction.getTime()
+                        sac.getSourceName(), sourceId, dac.getAdhocChannelId(), dac.getType(), interaction.getTime()
```
Same for line 699ff:
this.log.info(
                        "Added vehicle ID[int={}, ext={}] at position={} time={}",
                        simulatedNodes.fromExternalId(id), id, registeredNode.position, **TIME.format(time)**
);

fhilg replied:
done in https://github.com/eclipse/mosaic/pull/138/commits/13dbff3280132b74ded0334bed0236dae43794ad

fhilg replied:
whitespace removal done in https://github.com/eclipse/mosaic/pull/138/commits/672cd0419aa5b61c0ba9f665f3ca61bdb263523a

fhilg commented on lib/mosaic-network/src/main/java/org/eclipse/mosaic/lib/coupling/AbstractNetworkAmbassador.java:
```diff
@@ -318,13 +325,13 @@ protected void processTimeAdvanceGrant(long time) throws InternalFederateExcepti
                                     rcvMsgContainer.receiverInformation
                             );
-                            log.info("Receive V2XMessage : Id({}) on Node {} at Time={}", msg.getMessageId(), msg.getReceiverName(), msg.getTime());
+                            log.debug("Receive V2XMessage : Id({}) on Node {} at Time={}", msg.getMessageId(), msg.getReceiverName(), msg.getTime());
```
done in https://github.com/eclipse/mosaic/pull/138/commits/13dbff3280132b74ded0334bed0236dae43794ad

fhilg commented:
As discussed, I updated the ns-3 and OMNeT++ federate configurations, such that the new logging is applied and documented:

## ns-3 Federate
```
        <!-- These are setting suggestions for verbosity levels
              which are equivalent to MOSAIC's log level INFO and DEBUG

              MOSAIC log level equivalent:             INFO                                       DEBUG
         .....................................................................................................................-->
        <component name="MosaicStarter"        value="warn|error|debug|info"/>       <!-- value="warn|error|debug|info"      -->
        <component name="MosaicSimulatorImpl"  value="warn|error|debug|info"/>       <!-- value="warn|error|debug|info"      -->
        <component name="MosaicNs3Server"      value="warn|error"/>                  <!-- value="warn|error"                 -->
        <component name="MosaicNodeManager"    value="warn|error|debug"/>            <!-- value="warn|error|debug"           -->
        <component name="MosaicProxyApp"       value="warn|error|debug|func|info"/>  <!-- value="warn|error|debug|func|info" -->
        <component name="ClientServerChannel"  value="warn|error"/>                  <!-- value="warn|error"                 -->

        <component name="UdpSocketImpl"        value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="UdpL4Protocol"        value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="Ipv4L3Protocol"       value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="Ipv4Interface"        value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="TrafficControlLayer"  value="warn|error"/>                  <!-- value="warn|error|debug"           -->

        <component name="PropagationLossModel" value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="WifiPhy"              value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="YansWifiPhy"          value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="WaveNetDevice"        value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="YansWifiChannel"      value="warn|error"/>                  <!-- value="warn|error|debug"           -->
        <component name="InterferenceHelper"   value="warn|error"/>                  <!-- value="warn|error|debug"           -->
```

## OMNeT++ Federate
```
cmdenv-log-prefix = "%t: [%M] "

#...

# general logging output
# ----------------------

# These are setting suggestions for verbosity levels
# which are equivalent to MOSAIC's log level INFO and DEBUG
#
# MOSAIC log level equivalent:     INFO      DEBUG
#---------------------------------------------------
**.tx.cmdenv-log-level           = info   # = info
**.rx.cmdenv-log-level           = info   # = info
**.proxyApp.cmdenv-log-level     = info   # = info
Simulation.mgmt.cmdenv-log-level = info   # = info

**.cmdenv-log-level              = warn   # = info   # This sets everything to level INFO
```
